daemon/cmd/local_node_sync.go

```diff
@@ -52,7 +52,7 @@ func (ini *localNodeSynchronizer) InitLocalNode(ctx context.Context, n *node.Loc
 	if err := ini.initFromConfig(ctx, n); err != nil {
 		return err
 	}
-
+	// Deprecated: Restoring node information from k8s is deprecated as of v1.16.0 and should be removed in v1.17.0.
```

Comment:
This doesn't look correct to me. While I agree that CiliumInternalIPs cannot be restored from the Node object, this logic also takes care of initializing the LocalNodeStore with node Internal/External IP addresses as well as labels, annotations and a couple of extra fields, whose source of trust is the kubernetes Node object itself. This pieces of information are then eventually propagated into the corresponding CiliumNode object by the [node discovery logic](https://github.com/tommyp1ckles/cilium/blob/pr%2Ftp%2Frestore-cilium-host-ip-from-cn/pkg/nodediscovery/nodediscovery.go#L231-L236).

Reply:
You're right,  I think what I intended to say here is that we want to deprecate restoring cilium_host IP from k8s.   Saying we want to deprecate restoring from k8s is unfeasible and too broad.

Reply:
Going to remove this commit from now, I probably actually want to discuss this more.